### PR TITLE
feat!: support ThresholdRecoverRaw in tenderdash v0.14

### PR DESCRIPTION
**Pull request by lklimek**
## Issue being fixed or feature implemented

Tenderdash 0.14 contains new vote extension type, **ThresholdRecoverRaw**.

## What was done?

* added support for ThresholdRecoverRaw 
* allow selecting Tenderdash commit from repo by setting TENDERDASH_COMMITISH env variable
* fixed tests
* fixed clippy warnings
* added support and tests for ThresholdRecoverRaw signatures

## How Has This Been Tested?

Added unit test

## Breaking Changes

None

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: abci/src/signatures.rs [hunk 1]
```diff
@@ -33,18 +33,32 @@ const VOTE_EXTENSION_REQUEST_ID_PREFIX: &str = "dpevote";
 /// SignDigest returns message digest that should be provided directly to a
 /// signing/verification function (aka Sign ID)
 pub trait SignDigest {
+    #[deprecated = "replaced with sign_hash() to unify naming between core, platform and tenderdash"]
     fn sign_digest(
```
Comment (shumkov) [inline abci/src/signatures.rs:37]: both `sign_digest` and `sign_hash` are not the best names. I'm reading this as a function that signs a hash. 
Can we do signature_hash?

File: abci/src/signatures.rs [hunk 2]
```diff
@@ -318,20 +370,24 @@ impl SignBytes for CanonicalVote {
 
 impl SignBytes for VoteExtension {
     fn sign_bytes(&self, chain_id: &str, height: i64, round: i32) -> Result<Vec<u8>, Error> {
```
Comment (shumkov) [inline abci/src/signatures.rs:372]: The result of this function goes to sign_hash as sign_bytes_hash?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: abci/src/signatures.rs [hunk 3]
```diff
@@ -113,24 +133,62 @@ impl SignDigest for CanonicalVote {
...
             quorum_hash,
-            request_id[..].try_into().unwrap(),
+            request_id[..]
+                .try_into()
+                .expect("invalid request ID length"),
             &sign_bytes_hash,
-        ))
+        );
+
+        tracing::trace!(
```
Comment (shumkov) [inline abci/src/signatures.rs:184]: You are gonna remove it before merge right?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: abci/src/signatures.rs (hunk 3 above)
Comment (lklimek) [reply abci/src/signatures.rs:187]: added comment to remove once withdrawals are done

**Review by shumkov — APPROVED** [review]

**Pull request merged**